libs/vm/include/vm/variant.hpp

```diff
@@ -475,37 +475,23 @@ struct Variant
   }
 };
 
-struct TemplateParameter1 : public Variant
+template <std::size_t ID = 0>
+struct VariantT : public Variant
 {
   using Variant::Variant;
+  VariantT(Variant const &other)
```

Comment:
Should this be explicit?

Reply:
`VariantT` is capital T capitalised purposefully?

Reply:
The `Variant` base type does not have any constructor declared as explicit. All these types are, by design, just empty wrappers around `Variant` base type, with the only desired difference (when compared to Variant type itself) that they have different type id in C++ type system.